images/centos/7/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM docker.io/library/centos:7
+
+ENV NAME=centos-toolbox VERSION=7
+LABEL com.github.debarshiray.toolbox="true" \
+      com.redhat.component="$NAME" \
+      name="$FGC/$NAME" \
+      version="$VERSION" \
+      usage="This image is meant to be used with the toolbox command" \
+      summary="Base image for creating CentOS 7 toolbox containers" \
+      maintainer="Debarshi Ray <rishi@fedoraproject.org>"
+
+COPY README.md /
+
+RUN sed -i '/tsflags=nodocs/d' /etc/yum.conf
+#RUN yum -y swap coreutils coreutils-full
+RUN yum -y upgrade
+
+COPY missing-docs /
+RUN yum -y reinstall $(<missing-docs)
+RUN rm /missing-docs
+
+COPY extra-packages /
+RUN yum -y install $(<extra-packages)
+RUN rm /extra-packages
+
+RUN yum clean all
```

Comment:
Agreed. Actually I'm a bit surprised we aren't using buildah, but that sometimes makes things easier.

Reply:
> This command is useless when it's run within it's own layer,
> doesn't change anything other than making the resulting container
> go download everything.

Two things:

* The Fedora images are squashed (see `podman build --squash`), which makes it effective in [reducing the image sizes](https://github.com/containers/toolbox/commit/16f0b21ebc1de9562faad350002a79fecb4d49ef).
* Preserving the cache isn't useful because it's going to go stale pretty soon anyway, leaving us with a needlessly bigger image.

Reply:
> Actually I'm a bit surprised we aren't using buildah, but that sometimes
> makes things easier.

We use what the underlying infrastructure uses to produce the images. The Fedora infrastructure uses Dockerfiles.